protobuf/src/reflect/dynamic/mod.rs

```diff
@@ -186,19 +274,249 @@ impl Message for DynamicMessage {
     }
 
     fn is_initialized(&self) -> bool {
-        unimplemented!()
+        let fields: Vec<FieldDescriptor> = self.descriptor.fields().collect();
+        for f in &fields {
+            match f.runtime_field_type() {
+                RuntimeFieldType::Singular(rtb) => {
+                    if !self.check_singular_initialized(&rtb, &f) {
+                        return false;
+                    }
+                }
+                RuntimeFieldType::Repeated(rtb) => {
+                    if !self.check_repeated_initialized(&rtb, &f) {
+                        return false;
+                    }
+                }
+                RuntimeFieldType::Map(_, _) => {
+                    unimplemented!()
+                }
+            }
+        }
+
+        true
     }
 
-    fn merge_from(&mut self, _is: &mut CodedInputStream) -> ProtobufResult<()> {
-        unimplemented!()
+    fn merge_from(&mut self, is: &mut CodedInputStream) -> ProtobufResult<()> {
+        self.set_fields_default();
+        let desc = self.descriptor.clone();
+        while !is.eof()? {
+            let (field, wire_type) = is.read_tag_unpack()?;
+            let field_desc = desc
+                .get_field_by_number(field)
+                .expect("Invalid field number at decoding");
+            match field_desc.runtime_field_type() {
+                RuntimeFieldType::Singular(rtb) => {
+                    let val = match rtb {
+                        RuntimeTypeBox::I32 => ReflectValueBox::from(is.read_int32()?),
```

Comment:
This is incorrect: runtime `i32` can be serialized by several protobuf type: `int32`, `sint32`, `sfixed32`.

Reply:
If I understand it correctly, `conpute_size` and `write_to` also need to be modified accordingly?

Reply:
Yep.